src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -504,10 +504,13 @@ private void SendKeepAliveFrameAsync()
                 // This exists purely to keep the connection alive; don't wait for the result, and ignore any failures.
                 // The call will handle releasing the lock.
                 ValueTask t = SendFrameLockAcquiredNonCancelableAsync(MessageOpcode.Ping, true, Memory<byte>.Empty);
-
-                // "Observe" any exception, ignoring it to prevent the unobserved exception event from being raised.
-                if (!t.IsCompletedSuccessfully)
+                if (t.IsCompletedSuccessfully)
+                {
+                    t.GetAwaiter().GetResult();
```

Comment:
Not having this could result in our object reuse optimization getting defeated after the ping.